File: examples/collaborative-assistant/ui-module/ui_agent.py
import asyncio
from worldlink_client import Client

async def main():
    client = Client()
    await client.connect()

    agent_id = "ui_1"
    manifest = {
        "name": "UI Agent",
        "version": "1.0.0",
        "description": "Captures the user's query.",
        "capabilities": ["capture_query"]
    }
    await client.register_agent(agent_id, manifest)
    print(f"UI Agent '{agent_id}' registered.")

    await asyncio.sleep(2)

    user_query = "Can you generate a marketing plan for my AI-powered productivity app?"
    topic = "user_query"
    message = {"query": user_query}
    await client.publish_message(topic, message, sender_id=agent_id)
    print(f"UI Agent '{agent_id}' published user query: '{user_query}' to topic '{topic}'.")

    await asyncio.sleep(5)
    await client.disconnect()

if __name__ == "__main__":
    asyncio.run(main())

File: examples/collaborative-assistant/ui-module/ui_agent.js
const { Client } = require('worldlink-client');

async function main() {
    const client = new Client();
    await client.connect();

    const agentId = "ui_1";
    const manifest = {
        "name": "UI Agent",
        "version": "1.0.0",
        "description": "Captures the user's query.",
        "capabilities": ["capture_query"]
    };
    await client.registerAgent(agentId, manifest);
    console.log(`UI Agent '${agentId}' registered.`);

    await new Promise(resolve => setTimeout(resolve, 2000));

    const userQuery = "Can you generate a marketing plan for my AI-powered productivity app?";
    const topic = "user_query";
    const message = { query: userQuery };
    await client.publishMessage(topic, message, agentId);
    console.log(`UI Agent '${agentId}' published user query: '${userQuery}' to topic '${topic}'.`);

    await new Promise(resolve => setTimeout(resolve, 5000));
    await client.disconnect();
}

main();